src/versionUtils.ts

```diff
@@ -47,39 +47,44 @@ export async function testVersionAvailable(versionToUse: string): Promise<boolea
 }
 
 export async function checkKamelNeedsUpdate(versionToUse?: string): Promise<boolean> {
-	const runtimeVersionSetting = vscode.workspace.getConfiguration().get(config.RUNTIME_VERSION_KEY) as string;
-	if (versionToUse) {
-		const passedVersionAvailable = await testVersionAvailable(versionToUse);
-		if (!passedVersionAvailable) {
-			const msg: string = `Camel K CLI Version ${versionToUse} unavailable.`;
-			extension.shareMessageInMainOutputChannel(msg);
-			return false;
-		}
-	} else if (!versionToUse) {
-		const latestversion = await getLatestCamelKVersion();
-		if (failed(latestversion)) {
-			console.error(latestversion.error);
-			return true;
+	const runtimeVersionSetting = config.getKamelRuntimeVersionConfig();
+	const autoUpgradeSetting: boolean | undefined = config.getKamelAutoupgradeConfig();
+	const kamelCliToolsPath: string | undefined = config.getActiveKamelconfig();
```

Comment:
Just a minor thing but in the 2 lines above you are using the term `Setting` in the variable name while in this line you don't. Why not using the same terminology all over?